src/main_benchmark.cpp

```diff
@@ -0,0 +1,411 @@
+    const size_t niterPerThread,
+    size_t& wholeThreadTimeUs,
+    double& averageWholeLatency,
+    double& averagePureLatency,
+    OVMS_Server* server,
+    OVMS_InferenceRequest* requests) {  // TODO each thread should get 2 to switch
+    OVMS_InferenceResponse* response{nullptr};
+    startSignal.get();
+    std::vector<uint64_t> latenciesWhole(niterPerThread);
+    std::vector<uint64_t> latenciesPure(niterPerThread);
+    auto workloadStart = std::chrono::high_resolution_clock::now();
+    size_t iter = niterPerThread;
+    while (iter-- > 0) {
+        auto iterationStart = std::chrono::high_resolution_clock::now();
+        stopSignal.wait_for(std::chrono::milliseconds(0));
+        // TODO add here resetting buffer
+        auto iterationPureStart = std::chrono::high_resolution_clock::now();
+        // aternatively we are changing request
+        OVMS_Inference(server, requests, &response);
+        auto iterationPureEnd = std::chrono::high_resolution_clock::now();
+        OVMS_InferenceResponseDelete(response);
+        auto iterationEnd = std::chrono::high_resolution_clock::now();
+        latenciesWhole[iter] = std::chrono::duration_cast<std::chrono::microseconds>(iterationEnd - iterationStart).count();
+        latenciesPure[iter] = std::chrono::duration_cast<std::chrono::microseconds>(iterationPureEnd - iterationPureStart).count();
+    }
+    auto workloadEnd = std::chrono::high_resolution_clock::now();
+    wholeThreadTimeUs = std::chrono::duration_cast<std::chrono::microseconds>(workloadEnd - workloadStart).count();
+    auto result = std::accumulate(latenciesWhole.begin(), latenciesWhole.end(), 0);
+    averageWholeLatency = std::accumulate(latenciesWhole.begin(), latenciesWhole.end(), 0) / (double(niterPerThread) * 1'000);
+    averagePureLatency = std::accumulate(latenciesPure.begin(), latenciesPure.end(), 0) / (double(niterPerThread) * 1'000);
+}
+}  // namespace
+// TODO support more than 1 input
+// TODO support 3 modes:
+// * fair play -> don't recreate request, but set buffer inputs
+// * full request prepare -> recreate request
+// * inference only -> dont reset buffers
+// TODO change plugin_config
+// TODO change nireq in model
+
+int main(int argc, char** argv) {
+    installSignalHandlers();
+    BenchmarkCLIParser cliparser;
+    cliparser.parse(argc, argv);
+
+    OVMS_ServerGeneralOptions* go = 0;
+    OVMS_ServerMultiModelOptions* mmo = 0;
+    OVMS_Server* srv;
+
+    OVMS_ServerGeneralOptionsNew(&go);
+    OVMS_ServerMultiModelOptionsNew(&mmo);
+    OVMS_ServerNew(&srv);
+
+    uint32_t grpcPort = cliparser.result->operator[]("port").as<uint32_t>();
+    uint32_t restPort = cliparser.result->operator[]("rest_port").as<uint32_t>();
+    OVMS_ServerGeneralOptionsSetGrpcPort(go, grpcPort);
+    OVMS_ServerGeneralOptionsSetRestPort(go, restPort);
+
+    std::string cliLogLevel(cliparser.result->operator[]("log_level").as<std::string>());
+    OVMS_LogLevel_enum logLevel;
+    if (cliLogLevel == "TRACE") {
```

Comment:
This is sad that we make full circle of conversions:
`string -> enum` here
and then again `enum -> string` in C-API call